pkg/services/sqlstore/migrations/accesscontrol/team_membership.go

```diff
@@ -0,0 +1,152 @@
+package accesscontrol
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"xorm.io/xorm"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+func AddTeamMembershipMigrations(mg *migrator.Migrator) {
+	mg.AddMigration("teams permissions migration", &teamPermissionMigrator{})
+}
+
+var _ migrator.CodeMigration = new(teamPermissionMigrator)
+
+type teamPermissionMigrator struct {
+	migrator.MigrationBase
+}
+
+func (p *teamPermissionMigrator) SQL(dialect migrator.Dialect) string {
+	return "code migration"
+}
+
+func (p *teamPermissionMigrator) Exec(sess *xorm.Session, migrator *migrator.Migrator) error {
+	return p.migrateMemberships(sess)
+}
+
+// generateNewRoleUID tries to generate a unique Short UID three times, in case of three conflicts returns an error.
+func generateNewRoleUID(sess *xorm.Session, orgID int64) (string, error) {
```

Comment:
question: do we not have a function like this somewhere else that we could reuse?

Reply:
answer: yes, functions like this one are present all over the place, but there's nothing that can be used easily without a larger refactor.